lib/internal/util/embedding.js

```diff
@@ -36,12 +36,18 @@ function embedderRunCjs(contents) {
     customDirname);
 }
 
+let embedderRequireInternal;
+
 function embedderRequire(path) {
   if (!Module.isBuiltin(path)) {
     throw new ERR_UNKNOWN_BUILTIN_MODULE(path);
   }
 
-  return require(path);
+  if (!embedderRequireInternal) {
```

Comment:
If we are still not enabling fs I think we can just slice out the `node:` prefix like what the snapshot require does? https://github.com/nodejs/node/blob/77561d0e41752680aed1d4333b42a86b9870d2ce/lib/internal/main/mksnapshot.js#L101-L114

Reply:
@joyeecheung what would happen for something like the builtin test runner module where only `node:test` is the builtin module name and the one without the prefix, i.e., `test` isn't?

Reply:
That'd be covered by the `canBeRequiredWithoutScheme()` part.

Reply:
@joyeecheung done, PTAL. There was also a bug in the snapshot require logic while building the snapshot when `node:test` and `test` were being required. I've fixed that and updated the description with more info about it.